Makefile

```
@@ -1,17 +1,52 @@
+## Verify all generated code is fresh (CI + pre-push)
+gen-check: gen
+	@if ! git diff --quiet $(GENERATED_DIRS); then \
+		echo "ERROR: Generated code is stale. Run 'make gen' and commit."; \
+		git diff --stat $(GENERATED_DIRS); \
+		git checkout -- $(GENERATED_DIRS); \
+		exit 1; \
+	fi
+	@echo "✅ All generated code is fresh"
+
+## Validate local tool versions match VERSIONS file
+version-check:
```

checkmake (0.3.2)

[warning] Target body for "version-check" exceeds allowed length of 5 lines (10).

(maxbodylength)